saleor/checkout/fetch.py

```diff
@@ -548,7 +548,13 @@ def get_valid_internal_shipping_method_list_for_checkout_info(
     is_shipping_voucher = (
         checkout_info.voucher and checkout_info.voucher.type == VoucherType.SHIPPING
     )
-    if not is_shipping_voucher:
+
+    is_voucher_for_specific_product = (
+        checkout_info.voucher
+        and checkout_info.voucher.type == VoucherType.SPECIFIC_PRODUCT
+    )
```

Comment:
We need to also handle `apply_once_per_order` vouchers here as it is included in the subtotal too, AFAIK.

Reply:
We want to do that in a separate PR.